src/views/TabView/TabView.js

```diff
@@ -110,22 +110,37 @@ class TabView extends PureComponent<void, Props, void> {
 
   _renderTabBar = (props: *) => {
     const {
-      tabBarOptions,
-      tabBarComponent: TabBarComponent,
       animationEnabled,
+      navigation,
+      router,
+      screenProps,
+      tabBarComponent: TabBarComponent,
+      tabBarOptions,
     } = this.props;
     if (typeof TabBarComponent === 'undefined') {
       return null;
     }
+
+    const { state } = navigation;
+    const options = router.getScreenOptions(
+      this.props.childNavigationProps[state.routes[state.index].key],
+      screenProps || {}
+    );
+
+    const tabBarVisible =
+      options.tabBarVisible === null || options.tabBarVisible === undefined
+        ? true
+        : options.tabBarVisible;
+
     return (
       <TabBarComponent
         {...props}
         {...tabBarOptions}
-        screenProps={this.props.screenProps}
         navigation={this.props.navigation}
         getLabel={this._getLabel}
         renderIcon={this._renderIcon}
         animationEnabled={animationEnabled}
+        style={tabBarVisible ? undefined : styles.hidden}
```

Comment:
This clobbers the `style` prop in `tabBarOptions`. Fix:

```diff
diff --git a/src/views/TabView/TabView.js b/src/views/TabView/TabView.js
index 7cf6653..bae866a 100644
--- a/src/views/TabView/TabView.js
+++ b/src/views/TabView/TabView.js
@@ -13,12 +13,15 @@ import type {
   NavigationState,
   NavigationRouter,
   NavigationTabScreenOptions,
+  Style,
 } from '../../TypeDefinition';
 
 export type TabViewConfig = {
   tabBarComponent?: ReactClass<*>,
   tabBarPosition?: 'top' | 'bottom',
-  tabBarOptions?: {},
+  tabBarOptions?: {
+    style?: Style,
+  },
   swipeEnabled?: boolean,
   animationEnabled?: boolean,
   lazy?: boolean,
@@ -34,7 +37,9 @@ export type TabScene = {
 type Props = {
   tabBarComponent?: ReactClass<*>,
   tabBarPosition?: 'top' | 'bottom',
-  tabBarOptions?: {},
+  tabBarOptions?: {
+    style?: Style,
+  },
   swipeEnabled?: boolean,
   animationEnabled?: boolean,
   lazy?: boolean,
@@ -149,7 +154,7 @@ class TabView extends PureComponent<void, Props, void> {
         getOnPress={this._getOnPress}
         renderIcon={this._renderIcon}
         animationEnabled={animationEnabled}
-        style={!tabBarVisible ? styles.hidden : undefined}
+        style={[!tabBarVisible ? styles.hidden : undefined, tabBarOptions && tabBarOptions.style]}
       />
     )
   };
```

Reply:
Thanks for the review, already added